src/diagnostics/netcdf_writer.jl

```diff
@@ -138,11 +138,17 @@ function target_coordinates(
     S <:
     Union{Spaces.CenterFiniteDifferenceSpace, Spaces.FaceFiniteDifferenceSpace},
 }
+    # Logarithmically spaced with base e
+
     num_points_z = num_points[]
     FT = Spaces.undertype(space)
     vert_domain = space.topology.mesh.domain
-    z_min, z_max = vert_domain.coord_min.z, vert_domain.coord_max.z
-    return collect(range(FT(z_min), FT(z_max), num_points_z))
+    z_min, z_max = FT(vert_domain.coord_min.z), FT(vert_domain.coord_max.z)
+    # We are creating a log-space here, so we have to ensure that z_min > 0.
+    # If z_min = 0, we set the the first point to be at altitude of 100
+    log_z_min = z_min ≈ 0 ? 2 : log(z_min)
```

Comment:
```suggestion
    log_z_min = z_min ≈ 0 ? FT(2) : log(z_min)
```
(so that the result of `log_z_min` is type-stable)